birdhouse/optional-components/prometheus-longterm-metrics/prometheus.yml

```diff
@@ -0,0 +1,18 @@
+global:
+  external_labels:
+    instance_name: prometheus-longterm-metrics
+
+scrape_configs:
+  - job_name: 'federate'
+    scrape_interval: 15s
```

Comment:
It collects and stores metrics that are already collected by the other prometheus service. Depending on how often the other prometheus instance is calculating those metrics, you may not need such a small `scrape_interval`. What interval would you suggest?

Reply:
I made it configurable

Reply:
I do not fully understand federation yet but I had the impression the 2nd Prometheus would query the max/min/average value from the 1st Prometheus at a much longer interval.  Ex: the 1st Prometheus scrap every minute.  The 2nd Prometheus would scrap every hour, taking the max/min/average over the hour of the 1st Prometheus?

Reply:
You don't need multiple prometheus instances for that. You just need to create recording rules that calculate the metrics that you want to store for a longer period and then use the second prometheus instance to grab those metrics for long-term storage. 

The second prometheus instance is necessary only because the data retention time is configured for each instance, so if you want to keep data longer you need a separate prometheus instance.

If you're storing data for a long time with `thanos` instead, you still need a second prometheus instance so that you can select which data `thanos` will store. `thanos` stores **all** data from a prometheus instance, you cannot pick and choose, so we get around this problem by only collecting the data we are interested in the "longterm" prometheus instance and only using `thanos` to store data from that specific instance.

The best way to think about it is:

- `prometheus`: collects all metrics we are interested in and stores them for a short period of time
- `prometheus-longterm-metrics`: selects some metrics collected by `prometheus` that we are interested in storing for a long period of time.
- `thanos`: selects **all** metrics from `prometheus-longterm-metrics` and stores them efficiently over an even longer period of time

Reply:
I see clearer now, thanks !

Just to confirm, the current setup `prometheus-longterm-metrics` is collecting everything from `prometheus` so we are not yet "selecting some metrics" yet, right?

It's alright we can perform the selection in a subsequent PR.

Reply:
> Just to confirm, the current setup `prometheus-longterm-metrics` is collecting everything from `prometheus` so we are not yet "selecting some metrics" yet, right?

Opps sorry, just read your other reply now and so to answer my own remark:  `prometheus-longterm-metrics` is collecting only metrics from `group: longterm-metrics` from `prometheus`.